ArtifactResolutionRequest.setResolveTransitively(true) causes org.apache.maven.artifact.InvalidArtifactRTException

Hi,

I am facing the same issue as described here https://github.com/highsource/jaxb-tools/issues/653

```
Caused by: org.apache.maven.artifact.InvalidArtifactRTException: For artifact {com.gk-software.pos-platform:javapos-common:null:jar}: The version cannot be empty.
    at org.apache.maven.artifact.DefaultArtifact.validateIdentity (DefaultArtifact.java:162)
    at org.apache.maven.artifact.DefaultArtifact.<init> (DefaultArtifact.java:142)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createArtifact (DefaultArtifactFactory.java:177)
    at org.apache.maven.artifact.factory.DefaultArtifactFactory.createDependencyArtifact (DefaultArtifactFactory.java:66)
    at org.apache.maven.project.artifact.MavenMetadataSource.createDependencyArtifact (MavenMetadataSource.java:358)
    at org.apache.maven.project.artifact.MavenMetadataSource.createDependencyArtifact (MavenMetadataSource.java:340)
    at org.apache.maven.project.artifact.MavenMetadataSource.retrieve (MavenMetadataSource.java:265)
    at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:432)
    at org.apache.maven.repository.legacy.LegacyRepositorySystem.resolve (LegacyRepositorySystem.java:332)
    at org.jvnet.jaxb.maven.util.ArtifactUtils.resolveTransitively (ArtifactUtils.java:79)
    at org.jvnet.jaxb.maven.RawXJCMojo.resolveXJCPluginArtifacts (RawXJCMojo.java:397)
    at org.jvnet.jaxb.maven.RawXJCMojo.resolveArtifacts (RawXJCMojo.java:383)
    at org.jvnet.jaxb.maven.RawXJCMojo.execute (RawXJCMojo.java:320)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)

```

-----

@laurentschoelens said:
> Adding plugins to jaxb-maven-plugin will do classpath enhancements for external librairies discovery

-> But like this its quite bad... Is There a way to disable this behavior?

-----

- complex multi module project which I cannot change 
- artifact transitively included by other modules of the project 
- project is compiled and working so there basically is no problem with this, only this tool is causing problems 
- Dependency tree includes the artifact for different modules in the correct version

-----

- specifying this dependency explicitly in dependencies - same error occurs 
- specifying this dependency explicitly in dependency management - solves but next artifact is showing

-----

why is the plugin scanning the whole project artifacts anyway?
it should be enough to use the artifacts specified in plugin configuration ...

--- a/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/util/ArtifactUtils.java
+++ b/maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/util/ArtifactUtils.java
@@ -66,6 +66,7 @@ public class ArtifactUtils {
 
         ArtifactResolutionRequest request = new ArtifactResolutionRequest();
         request.setResolveTransitively(true);
+        request.setManagedVersionMap(Collections.emptyMap());
         request.setResolveRoot(false);
         request.setArtifact(project.getArtifact());
         request.setArtifactDependencies(artifacts);

--- a/maven-plugin/tests/episodes/c/pom.xml
+++ b/maven-plugin/tests/episodes/c/pom.xml
@@ -42,6 +42,13 @@
             </catalog>
           </catalogs>
           <useDependenciesAsEpisodes>true</useDependenciesAsEpisodes>
+          <plugins>
+            <plugin>
+              <groupId>org.jvnet.jaxb</groupId>
+              <artifactId>jaxb-plugins</artifactId>
+              <version>${project.version}</version>
+            </plugin>
+          </plugins>
         </configuration>
       </plugin>
     </plugins>

--- a/maven-plugin/tests/episodes/pom.xml
+++ b/maven-plugin/tests/episodes/pom.xml
@@ -16,6 +16,14 @@
     <module>d</module>
     <module>e</module>
   </modules>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>listenablefuture</artifactId>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
   <dependencies>
     <dependency>
       <groupId>org.glassfish.jaxb</groupId>

